docs/site/Fields-filter.md

```diff
@@ -0,0 +1,83 @@
+---
+title: 'Fields filter'
+lang: en
+keywords: LoopBack 4.0, LoopBack 4, fields
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Fields-filter.html
+---
+
+A _fields_ filter specifies properties (fields) to include or exclude from the results.
+
+### Node.js API
+
+{% include content/angular-methods-caveat.html lang=page.lang %}
+
+<pre>
+{ fields: {<i>propertyName</i>: <true|false>, <i>propertyName</i>: <true|false>, ... } }
+</pre>
+
+Where:
+
+- _propertyName_ is the name of the property (field) to include or exclude.
+- `<true|false>` signifies either `true` or `false` Boolean literal. Use `true` to include the property or `false` to exclude it from results.
+
+By default, queries return all model properties in results. However, if you specify at least one fields filter with a value of `true`,
+then by default the query will include **only** those you specifically include with filters.
+
+### REST API
+
+{% include warning.html content="
+As a known bug, the option `<false>` does not work for url for now. Please include the properties you need to meet your requirement. The bug is tracked in the GH issue [#4992](https://github.com/strongloop/loopback-next/issues/4992" %}
+
+<pre>
+filter[fields][<i>propertyName</i>]=<true>&filter[fields][<i>propertyName</i>]=<true>...
+</pre>
+
+Note that to include more than one field in REST, use multiple filters.
+
+You can also use [stringified JSON format](Querying-data.html#using-stringified-json-in-rest-queries) in a REST query.
+
+### Examples
+
+Return customer information only with `name` and `address`, and hide their `id`:
+
+{% include code-caption.html content="Node.js API" %}
+
+```ts
+await customerRepository.find({fields: {id: false, name: true, address: true}});
+```
+
+{% include code-caption.html content="REST" %}
+
+Include all required properties as a workaround:
+`/customers?filter[fields][name]=true&filter[fields][address]=true`
+
+Returns:
+
+```ts
+[
+  {
+    name: 'Mario',
+    address: '8200 Warden Ave'
+  }, 
+  {
+    name: 'Luigi',
+    address: '999 Avenue Rd'
+  },
+  ...
+]
+```
+
+Exclude the `password` property:
+
+{% include code-caption.html content="Node.js API" %}
+```ts
+await userRepository.find({fields: {password: false}});
+```
+
+{% include code-caption.html content="REST" %}
+
+Include all properties except `password` as a workaround:
+`/users?filter[fields][name]=true&filter[fields][email]=true&filter[fields][id]=true...`
```

Comment:
Please include a note about how some properties can be hidden by default, else users might end up "protecting" sensitive data with a `false` flag in the query, which can easily be circumvented - https://github.com/strongloop/loopback-next/pull/3304. For more context and discussion - https://github.com/strongloop/loopback-next/issues/1914.

Reply:
👍 agree, it is included in https://github.com/strongloop/loopback-next/pull/4970/files#diff-d205b8643b362fb2063b843f87a94dd9R83 and is pointing to the doc https://loopback.io/doc/en/lb4/Model.html#hidden-properties